core/src/main/java/de/fraunhofer/iosb/ilt/faaast/service/persistence/util/EnvironmentHelper.java

```diff
@@ -61,8 +63,35 @@ public static List<AssetAdministrationShell> getDeepCopiedShells(Predicate<Asset
      * @param <T> type of the identifiable
      */
     public static <T extends Identifiable> void updateIdentifiableList(List<T> list, Identifiable identifiable) {
-        list.removeIf(x -> x.getIdentification().getIdentifier().equalsIgnoreCase(identifiable.getIdentification().getIdentifier()));
-        list.add((T) identifiable);
+        Identifiable actualIdentifiable = list.stream()
+                .filter(x -> x.getIdentification().getIdentifier().equalsIgnoreCase(identifiable.getIdentification().getIdentifier()))
+                .findFirst()
+                .orElse(null);
+
+        int index = list.indexOf(actualIdentifiable);
+        list.remove(actualIdentifiable);
+        add(list, index, (T) identifiable);
+    }
+
+
+    /**
+     * Adds the element to the collection. If the concrete collection supports adding an element at a specific index
+     * the element will be added at the given index.
+     * If not, the element is added at the end of the collection.
+     *
+     * @param collection to add the element
+     * @param index where to add the element in the list. Ignored when negative.
+     * @param element to add
+     * @param <T> type of the element
+     */
+    public static <T> void add(Collection<T> collection, int index, T element) {
```

Comment:
move this to new class CollectionHelper as it is independant of context and may be reused elsewhere